connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp

```diff
@@ -0,0 +1,1164 @@
+    }
+}
+
+const char *RIL_CellularContext::get_nonip_context_type_str()
+{
+    return nullptr;
+}
+
+void RIL_CellularContext::request_setup_data_call_response(ril_token_t *token, RIL_Errno err, void *response,
+                                                           size_t response_len)
+{
+    if (err != RIL_E_CANCELLED) {
+        token->cond_mutex->lock();
+        if (err == RIL_E_SUCCESS && response && response_len) {
+            RIL_Data_Call_Response_v11 *v11;
+            // only one response in RIL_REQUEST_SETUP_DATA_CALL
+            v11 = (RIL_Data_Call_Response_v11 *)response;
+            _cb_data.error = NSAPI_ERROR_OK;
+            if (v11->status == PDP_FAIL_NONE && v11->active == 2) {
+                _cid = v11->cid;
+
+                _pdp_type = string_to_pdp_type(v11->type);
+                tr_debug("RIL_CellularContext::request_setup_data_call_response, cid: %d, pdp_type: %s", _cid, v11->type);
+
+                if (v11->ifname) {
+                    const uint32_t len = strlen(v11->ifname) + 1;
+                    delete [] _ifname;
+                    _ifname = new char[len];
+                    memcpy(_ifname, v11->ifname, len);
+                    tr_debug("RIL_CellularContext::request_setup_data_call_response, _ifname: %s", _ifname);
+                } else {
+                    _cb_data.error = NSAPI_ERROR_DEVICE_ERROR;
+                }
+
+                // Skip address fields in case of NonIP context
+                if (_pdp_type != NON_IP_PDP_TYPE && !_cb_data.error) {
+                    _cb_data.error = handle_data_call_setup_addresses(v11);
+                } else {
+                    tr_debug("Skipping address fields as not needed for NonIP context");
+                }
+
+                if (v11->mtu && !_cb_data.error) {
+                    _mtu = v11->mtu;
+                    tr_debug("RIL_CellularContext::request_setup_data_call_response, _mtu: %d", _mtu);
+                }
+            } else {
+                tr_error("RIL_CellularContext::request_setup_data_call_response fail: status = %d, active = %d",
+                         v11->status, v11->active);
+                _cb_data.error = NSAPI_ERROR_DEVICE_ERROR;
+            }
+        } else {
+            tr_error("RIL_CellularContext::request_setup_data_call_response fail: err = %d, response = %p, response_len = %d",
+                     err, response, response_len);
+            _cb_data.error = NSAPI_ERROR_DEVICE_ERROR;
+        }
+
+        token->response_error = _cb_data.error;
+        token->cond_var->notify_one();
+        token->cond_mutex->unlock();
+    }
+}
+
+nsapi_error_t RIL_CellularContext::handle_data_call_setup_addresses(const RIL_Data_Call_Response_v11 *response)
+{
+    char temp[64];
+
+    delete _data_call_addresses;
+    _data_call_addresses = new data_call_addr_data_t;
+
+    if (response->addresses) {
+        mbed_cellular_util::separate_ip_addresses(response->addresses,
```

Comment:
These are space delimited lists, so should be
`mbed_cellular_util::separate_space_delimited_addresses(true, response->addresses,`

Reply:
This PR is based on mbed os 6 to add RIL interface, the pr-S5JS100-06_RIL is based on mbed os 5, so there is no `void separate_space_delimited_addresses(bool ipv6,char *orig,char *ip1, size_t ip1_size,char *ip2, size_t ip2_size)` function in the CellularUtil.h
Add our principle in the adaptation process is not to change your commit code.

Reply:
`separate_space_delimited_addresses` is a new functionality that needs to be added to CellularUtility class. `separate_ip_addresses` does not work because in this case as the input format is incompatible with this one.

Reply:
OK, I got it and will to add it, thanks